File: week1/승진/BJ1890_점프.py
import sys

input = sys.stdin.readline

N = int(input().rstrip())
maps = []
result = 0
for _ in range(N):
    maps.append(list(map(int,input().rstrip().split())))

dp = [[0 for _ in range(N)] for _ in range(N)]

dp[0][0] = 1 

for i in range(N):
    for j in range(N):

        if i == N - 1 and j == N - 1:
            print(dp[i][j])
            break

        if j + maps[i][j] < N:
            dp[i][j + maps[i][j]] += dp[i][j]

        if i + maps[i][j] < N:
            dp[i + maps[i][j]][j] += dp[i][j]

File: week1/승진/BJ1890_점프.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int[][] maps = new int[N][N];
        long[][] dp = new long[N][N];  // long 자료형으로 변경

        for (int i = 0; i < N; i++) {
            String[] mapRow = br.readLine().split(" ");
            for (int j = 0; j < N; j++) {
                maps[i][j] = Integer.parseInt(mapRow[j]);
            }
        }

        dp[0][0] = 1;

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                if (i == N - 1 && j == N - 1) {
                    System.out.println(dp[i][j]);
                    return;  // 루프 종료 후 바로 리턴
                }

                if (j + maps[i][j] < N) {
                    dp[i][j + maps[i][j]] += dp[i][j];
                }

                if (i + maps[i][j] < N) {
                    dp[i + maps[i][j]][j] += dp[i][j];
                }
            }
        }
    }
}
